### PR TITLE
fix strHashMap bug

### PR DESCRIPTION
**Pull request by JackTan25**
## What type of PR is this?

- [ ] API-change
- [x] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #7246

## What this PR does / why we need it:

### CODE REVIEW

**Review by aressu1985 — APPROVED** [review]

**Review by fengttt — COMMENTED** [review]
this is not going to work.   how can you distinguish `"a\0", "b"` and `"a", "\0b"`?

File: pkg/common/hashmap/strhashmap.go [hunk 1]
```diff
@@ -170,15 +171,36 @@ func (m *StrHashMap) encodeHashKeys(vecs []*vector.Vector, start, count int) {
 	}
 }
 
-func fillStringGroupStr(m *StrHashMap, vec *vector.Vector, n int, start int) {
+func HashKeys(keys []byte, bytes []byte) []byte {
```
Comment (fengttt) [inline pkg/common/hashmap/strhashmap.go:174]: no, you are not going to use make here.  

you are just serialize keys to a bytes, correct?   why not just length prefix + data?   You should compute the length you need, call one append at most.

File: pkg/common/hashmap/strhashmap.go [hunk 2]
```diff
@@ -170,15 +171,36 @@ func (m *StrHashMap) encodeHashKeys(vecs []*vector.Vector, start, count int) {
...
 	vs := vector.MustTCols[types.Varlena](vec)
 	if !vec.GetNulls().Any() {
 		for i := 0; i < n; i++ {
 			if m.hasNull {
 				m.keys[i] = append(m.keys[i], byte(0))
 			}
-			m.keys[i] = append(m.keys[i], vs[i+start].GetByteSlice(area)...)
+			// for "a"，"bc" and "ab","c", we need to distinct
+			if lenCols > 1 {
+				bytes := make([]byte, 24)
```
Comment (fengttt) [inline pkg/common/hashmap/strhashmap.go:198]: every col you call make?   the speed and gc overhead is not acceptable.

File: pkg/common/hashmap/strhashmap.go (hunk 1 above)
Comment (JackTan25) [reply pkg/common/hashmap/strhashmap.go:174]: the hash function need the len(keys) at least 16bytes, but I can't modify the source keys, however I will give you another way

**Review by aunjgr — APPROVED** [review]
lgtm

**Review by heni02 — APPROVED** [review]

**Comment by JackTan25** [comment]
@matrix-meow rerun failed

**Comment by JackTan25** [comment]
@matrix-meow rerun failed

**Review by nnsgmsone — APPROVED** [review]

**Comment by JackTan25** [comment]
@matrix-meow rerun failed

**Pull request merged**